snapcraft.yaml

```diff
@@ -14,23 +14,48 @@ architectures:
   - build-on: arm64
   - build-on: armhf
 
+plugs:
+  sys-devices-virtual-info:
+    interface: system-files
+    read:
+    # tailscale uses this to figure out if running in VM and if so, which Public Cloud
+    - /sys/devices/virtual/dmi/id/bios_vendor
+    - /sys/devices/virtual/dmi/id/product_name
+
 apps:
   tailscale:
     command: bin/tailscale
+    plugs:
+      - network
+      - network-observe
   tailscaled:
-    command: bin/tailscaled --statedir $SNAP_DATA
+    command: bin/tailscaled --statedir $SNAP_DATA --verbose 10
     daemon: simple
     plugs:
       - firewall-control
       - network
       - network-bind
+      - network-manager
       - network-control
+      - network-observe
+      - sys-devices-virtual-info
+      - system-observe
 
 parts:
   tailscale:
     source: https://github.com/tailscale/tailscale.git
     source-tag: v$SNAPCRAFT_PROJECT_VERSION
     plugin: go
+    build-snaps:
+      - go
+    stage-packages:
+      - iptables
+      - openssh-client # for tailscale ssh
     override-build: |
       sed -i paths/paths.go -e 's|"/var/run/tailscale/tailscaled.sock"|filepath.Join(os.Getenv("SNAP_DATA"), "tailscaled.sock")|'
-      snapcraftctl build
+      go install -p $CRAFT_PARALLEL_BUILD_COUNT ./cmd/tailscale
+      go install -p $CRAFT_PARALLEL_BUILD_COUNT ./cmd/tailscaled
```

Comment:
why is this needed now?

Reply:
Because we don't need to build all possible binaries. Those two are the ones that are shipped /w other packages as well.